index.js

```diff
@@ -107,18 +107,34 @@ export function checkDataIntegrityProofFormat({
               }
             });
         }
-        it('"proof.created" field MUST exist and be a valid XMLSCHEMA-11 ' +
+        it('if "proof.created" field exists, it MUST be a valid XMLSCHEMA-11 ' +
           'datetime value.', function() {
           this.test.cell = {columnId: vendorName, rowId: this.test.title};
           for(const proof of proofs) {
-            proof.should.have.property('created');
-            // check if "created" is a valid ISO 8601 datetime value
-            const dateRegex = new RegExp('^(\\d{4})-(0[1-9]|1[0-2])-' +
-            '(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):' +
-            '([0-5][0-9]):([0-5][0-9]|60)' +
-            '(\\.[0-9]+)?(Z|(\\+|-)([01][0-9]|2[0-3]):' +
-            '([0-5][0-9]))$', 'i');
-            proof.created.should.match(dateRegex);
+            if(proof.created) {
+              // check if "created" is a valid ISO 8601 datetime value
+              const dateRegex = new RegExp('^(\\d{4})-(0[1-9]|1[0-2])-' +
+              '(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):' +
+              '([0-5][0-9]):([0-5][0-9]|60)' +
+              '(\\.[0-9]+)?(Z|(\\+|-)([01][0-9]|2[0-3]):' +
+              '([0-5][0-9]))$', 'i');
+              proof.created.should.match(dateRegex);
+            }
+          }
+        });
+        it('if "proof.expires" field exists, it MUST be a valid XMLSCHEMA-11 ' +
+          'datetime value.', function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          for(const proof of proofs) {
+            if(proof.expires) {
+              // check if "created" is a valid ISO 8601 datetime value
+              const dateRegex = new RegExp('^(\\d{4})-(0[1-9]|1[0-2])-' +
+              '(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):' +
+              '([0-5][0-9]):([0-5][0-9]|60)' +
+              '(\\.[0-9]+)?(Z|(\\+|-)([01][0-9]|2[0-3]):' +
+              '([0-5][0-9]))$', 'i');
+              proof.expires.should.match(dateRegex);
+            }
```

Comment:
Are the dateRegex values exactly the same? If so, create the regex once and use it in both places. Avoid copy-pasting code in the same file... doing so can eventually leads to errors where you fix one regex and forget to fix the other one.

Reply:
Makes sense. Updated here [1300ee1](https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/16/commits/1300ee178696371f535eb7f80c9295f9c814db1e)